### PR TITLE
[ONNX] Remove spammy constant folding unsupported version warnings

### LINKED ISSUE #85015: [ONNX] Remove spammy warnings
- [x] `Warning: Constant folding - unsupported opset version. Constant folding not applied.`(https://github.com/pytorch/pytorch/pull/89291)
- [x] `WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.`(https://github.com/pytorch/pytorch/pull/88622)

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89291

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/constant_fold.cpp [hunk 1]
```diff
@@ -341,11 +339,6 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
         updated_val = at::unsqueeze(updated_val, axis);
       }
       return c10::optional<at::Tensor>(updated_val);
-    } else {
-      TORCH_WARN(
-          "Constant folding - unsupported opset version. "
-          "Constant folding not applied.");
-      return c10::nullopt;
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/constant_fold.cpp:348]: Is it ok to remove the return statement too?

Comment (AllenTiTaiWang) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp:348]: I will change the else if to else.

Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp:348]: I suppose the early return logic still needs to be kept?

Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp:348]: Interesting. If we are observing spammy constant folding warning, that must imply the guarded else branch of `opset < 9` is hit, right? Another bug somewhere else?

Comment (AllenTiTaiWang) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp:348]: Nooooooo

**Comment by AllenTiTaiWang** [comment]
I can't reproduce this in any case. The warning was introduced way back to #22515. I guess it was intended to be double check for the version, and then developers are just stacking up the code. As long as this is not creating issues, and blocking anyone, I think we can ignore it for now. Solving this simply benefits to opset <= 9 which isn't even the main opset versions we support anymore. But fixing it might break something that we don't know about yet. Close this PR and issue.

**Pull request closed without merging**